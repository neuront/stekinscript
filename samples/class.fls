class Animal
    ctor(name)
        this.name: name

    func move(x)
        return this.name + ' moved ' + x + ' meters'

class Bird: Animal
    ctor(name, speed) super(name)
        this.speed: speed

    func move(x)
        if this.speed = 1
            return 'Bird ' + super.move(x)
        return 'Bird ' + this.name + ' flied ' + x * this.speed + ' meters'

a: Animal('Alche')
console.log(a.move(5))

b: Bird('Rokh', 2)
console.log(b.move(4))

d: Bird('Duck', 1)
console.log(d.move(3))
