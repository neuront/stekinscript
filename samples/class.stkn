class Animal
    ctor (name)
        @name: name

    func move(x)
        return @name + ' moved ' + x + ' meters'

a: Animal('Alche')
console.log(a.move(5))
