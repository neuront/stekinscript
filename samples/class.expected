Alche moved 5 meters
